src/lib/components/UrlChooserContainer/index.js

```diff
@@ -0,0 +1,78 @@
+import {html} from "lit-element";
+import {store} from "../../store";
+import {BaseElement} from "../BaseElement";
+import {userRef, firebase} from "../../fb";
+import "../UrlChooser";
+
+/**
+ * @fileoverview Manages state interaction with UrlChooser.
+ *
+ * Invokes Lighthouse when the UrlChooser requests it, possibly with an updated URL.
+ */
+
+/* eslint-disable require-jsdoc */
+class UrlChooserContainer extends BaseElement {
+  static get properties() {
+    return {
+      url: {type: String},
+      active: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.url = null; // when signed out or waiting for Firestore, this is null
+    this.active = false;
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+  }
+
+  render() {
+    return html`
+      <web-url-chooser
+        .url=${this.url}
+        .active=${this.active}
+        @audit=${this.runAudit}
+      ></web-url-chooser>
+    `;
+  }
+
+  onStateChanged() {
+    const state = store.getState();
+
+    // if Lighthouse is active, ignore changes to the URL from Firestore
+    this.url = state.activeLighthouseUrl || state.userUrl;
+    this.active = state.activeLighthouseUrl !== null;
+  }
+
+  runAudit(e) {
+    const url = e.detail;
+
+    // write URL to firestore and local
+    // TODO(samthor): This should happen in a controller setting.
+    store.setState({
+      userUrl: url,
```

Comment:
Can you help me understand the difference between `userUrl` and `activeLighthouseUrl`?